core/src/operations/mod.rs

```diff
@@ -381,6 +381,9 @@ pub enum Operation {
     /// This operation affects only the advice tape, but has no effect on other VM components
     /// (e.g., stack, memory), and does not advance VM clock.
     Advice(AdviceInjector),
+
+    /// Added to the top of VmStateIterator
+    None,
```

Comment:
I'm not too sure about having this here. It isn't really needed for VM execution - only for debug purposes. I was thinking that we'd change `VmState` to something like:
```Rust
pub struct VmState {
    pub clk: usize,
    pub op: Option<Operation>,
    pub fmp: Felt,
    pub stack: Vec<Felt>,
    pub memory: Vec<(u64, Word)>,
}
```
And handle `None` at that level.

Reply:
Thanks. I was unsure of what the right way to handle it was out of the 2 (introducing variant or changing type to Option). I've changed to Option based on your suggestion in the latest commit. Makes sense since it is not required in non-debug mode.